test/acceptance/features/bindAppToService.feature

```diff
@@ -804,3 +804,103 @@ Feature: Bind an application to a service
             """
         Then Error message "spec: Required value" is thrown
         And Service Binding "binding-request-remove-spec" is not updated
+
+
+   Scenario: Backend Service metadata annotations update for service bindings gets propagated to the binding secret
+        Given OLM Operator "backend" is running
+        * The Custom Resource is present
+            """
+            apiVersion: "stable.example.com/v1"
+            kind: Backend
+            metadata:
+                name: backend-demo-2
+                annotations:
+                    service.binding/host: path={.spec.host}
+            spec:
+                host: example.com
+            status:
+                ready:true
+            """
+        * Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-backend-ann-sb
+            spec:
+                services:
+                -   group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: backend-demo-2
+            """
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "binding-request-backend-ann-sb" should be changed to "True"
```

Comment:
use `And` because we are still in the preparation phase here. I would even say that is not really required to check the content of the secret, because we are interested to see what is going to happen after update.

Reply:
Retriggering of the reconcile can be seen from the changes in the secret's data. Something like what was followed here https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/bindAppToService.feature#L277  We are concerned with the secret contents to be updated , not exactly how the app users will use the secret. hence for if we are interested to see what is going to happen after update, it means to check the secret contents. Please let me know what you suggest here. According to my this checking is required.

Reply:
My point was that these checks could be dropped because you do the final check at the end - what happened inbetween is not super important - only what happens at the end counts :) but I am ok if you want to leave them.